packages/knit/package.json

```diff
@@ -10,19 +10,34 @@
   },
   "sideEffects": false,
   "type": "module",
-  "types": "./dist/index.d.ts",
+  "main": "./dist/cjs/index.js",
+  "module": "./dist/esm/index.js",
   "exports": {
     ".": {
       "import": "./dist/esm/index.js",
-      "require": "./dist/index.js"
+      "require": "./dist/cjs/index.js"
+    },
+    "./gateway": {
+      "import": "./dist/esm/gateway/index.js",
+      "require": "./dist/cjs/gateway/index.js"
+    }
+  },
+  "typesVersions": {
+    "*": {
+      "gateway": [
+        "./dist/esm/gateway/index.d.ts",
+        "./dist/cjs/gateway/index.d.ts"
+      ]
     }
   },
   "scripts": {
     "clean": "rm -rf ./dist/* .turbo/*",
     "lint": "eslint .",
     "test": "NODE_OPTIONS=--experimental-vm-modules ../../node_modules/.bin/jest",
     "test:watch": "pnpm run build:esm+types --watch & pnpm run test --watchAll",
-    "build": "tsup src/index.ts --dts --no-splitting --format esm,cjs --legacy-output --treeshake"
+    "build:esm": "tsup src/index.ts src/gateway/index.ts --dts --no-splitting --legacy-output --clean --treeshake --outDir ./dist/esm --format esm && cp package.esm.json ./dist/esm/package.json",
```

Comment:
unfortunately, since `cp` is not cross-platform, using it here means that will make knit also not be cross-platform.  your best bet is a node-script in my opinion.  then, at least, you also can avoid making three package.json files and instead and just change the fields you need but copy the rest over.  see https://github.com/bufbuild/connect-query/pull/20/files#diff-c32299e3911888dd115eded04a2f4b9a070d82c28a58c34de22c9f7911527d73 for an example of when we tried doing the exact same thing for connect-query (but ended up going a different direction for other reasons).

Reply:
Moved everything to `tsup.config.ts` and also added the legacy bundler workaround for sub-path exports that connect uses.

Reply:
great stuff